### PR TITLE
RandomPractice

### PR DESCRIPTION
**Pull request by duhan9836**

### CODE REVIEW

File: RandomPractice.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+def generateOne(n):
+    alphabet="abcdefghijklmnopqrstuvwxyz "
+    return random.choices(alphabet,k=n)
+#the first method returns a list of letters
+
+def generateTwo(n):
+    alphabet="abcdefghijklmnopqrstuvwxya "
+    teststring=""
+    for i in range(n):
+        teststring=teststring+alphabet[random.randrange(len(alphabet))]
```
Comment (caitao1985) [inline RandomPractice.py:13]: can be written as:

```python
alphabet="abcdefghijklmnopqrstuvwxya "
return "".join( (alphabet[random.randrange(len(alphabet))] for i in range(n) ) )
```

File: RandomPractice.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    return teststring
+#the second method returns a string
+
+#define a function to score a generated string, by comparing the goalstring and teststring
+def score(goalstring,teststring):
+    samesum=0
+    for i in range(len(goalstring)):
+        if goalstring[i]==teststring[i]:
+            samesum+=1
```
Comment (caitao1985) [inline RandomPractice.py:22]: can be re-written as:

```python
return sum( (1 if (goalstring[i]==teststring[i]) else 0  for i in range(len(goalstring)) ））／ len(goalstring)

```

File: RandomPractice.py (hunk 1 above)
Comment (duhan9836) [reply RandomPractice.py:13]: brilliant! haven't thought of using list comprehension in such circumstances. but you use tuple instead of list?

Comment (caitao1985) [reply RandomPractice.py:13]: good question, but this is not tuple.

**Pull request merged**